### ISSUE TITLE
Bug with writing out doubles with toml_writer

### ISSUE THREAD

**Issue by dcdillon** [skystrife/cpptoml#15]
Doubles that are integers (i.e. 3.0) don't get a decimal point when written out by toml_writer which will cause the TOML to be invalid in the case where they were members of an array of doubles.  A reasonable fix is below, though you may want to do some saving and restoring of ostream flags as well.  I'd do it myself but I'm knee deep in something else.  There is also another issue with the precision of doubles being written out...currently it's whatever the stream's current setting is.  We may want to do something about that.

```c++
/**
 * Write out a double.
 */
void write(const value<double>& v)
{
    stream_ << std::showpoint;
    write(v.get());
}
```

**Comment by skystrife** [comment]
See my comment [here](https://github.com/skystrife/cpptoml/pull/16#issuecomment-137988025). I think we can do this with something like

```cpp
void write(const value<double>& v)
{
    // save old precision and flags
    std::ios::fmtflags flags{stream_.flags()};
    auto precision = stream_.precision();

    // ensure that (1) the dot is written and (2) we write as many digits as we need to
    stream_ << std::showpoint 
            << std::setprecision(std::numeric_limits<double>::max_digits10);
    write(v.get());

    // restore the old stream state
    stream_.flags(flags);
    stream_.precision(precision);
}
```

so that we properly respect the stream's state before and after the call.

**Comment by skystrife** [comment]
Fixed in 9b53ef7.
